core/plugin_api.rs

```diff
@@ -20,4 +22,15 @@ pub type DispatchOpFn =
 
 pub trait Interface {
   fn register_op(&mut self, name: &str, dispatcher: DispatchOpFn) -> OpId;
+  fn resource_table(&mut self) -> &mut dyn ResourceTable;
+}
+
+/// Similar to `struct ResourceTable` for normal ops, but uses dynamic dispatch
+/// instead of type parameters for its methods.
+pub trait ResourceTable {
+  fn add(&mut self, name: &str, resource: Box<dyn Resource>) -> ResourceId;
+  fn close(&mut self, rid: ResourceId) -> Option<()>;
+  fn get(&self, rid: ResourceId) -> Option<&dyn Resource>;
+  fn get_mut(&mut self, rid: ResourceId) -> Option<&mut dyn Resource>;
```

Comment:
Can these two methods take type parameter and perform downcast like `core::ResourceTable`? Then APIs would be identical

Reply:
They cannot. See comment at top of the file.

Reply:
I do think that the real resource table could implement this trait with almost no effort tho.